python_modules/dagster/dagster_tests/storage_tests/utils/event_log_storage.py

```diff
@@ -5124,6 +5124,11 @@ def _occupy_slot(key: str):
             while not all(f.done() for f in futures) and time.time() < start + TOTAL_TIMEOUT_TIME:
                 time.sleep(0.1)
 
+            exceptions = [f.exception() for f in futures]
+            exceptions = [ex for ex in exceptions if ex]
+            if exceptions:
+                raise exceptions[0]
```

Comment:
i think can just call  `future.result()` and it will re-raise any exceptions 
or use `executor.map` with `timeout` instead of `submit`
https://docs.python.org/3/library/concurrent.futures.html